codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/customizations/RulesBasedAuthSchemeResolverGeneratorTests.kt

```diff
@@ -0,0 +1,151 @@
+                    var validAuthOptions = [AuthOption]()
+                    guard let serviceParams = params as? S3AuthSchemeResolverParameters else {
+                        throw ClientError.authError("Service specific auth scheme parameters type must be passed to auth scheme resolver.")
+                    }
+                    let endpointParams = EndpointParams(authSchemeParams: serviceParams)
+                    let endpoint = try DefaultEndpointResolver().resolve(params: endpointParams)
+                    guard let authSchemes = endpoint.authSchemes() else {
+                        return try InternalModeledS3AuthSchemeResolver().resolveAuthScheme(params: params)
+                    }
+                    let schemes = try authSchemes.map { (input) -> AWSClientRuntime.AuthScheme in try AWSClientRuntime.AuthScheme(from: input) }
+                    for scheme in schemes {
+                        switch scheme {
+                            case .sigV4(let param):
+                                var sigV4Option = AuthOption(schemeID: "aws.auth#sigv4")
+                                sigV4Option.signingProperties.set(key: AttributeKeys.signingName, value: param.signingName)
+                                sigV4Option.signingProperties.set(key: AttributeKeys.signingRegion, value: param.signingRegion)
+                                validAuthOptions.append(sigV4Option)
+                            case .sigV4A(let param):
+                                var sigV4Option = AuthOption(schemeID: "aws.auth#sigv4a")
+                                sigV4Option.signingProperties.set(key: AttributeKeys.signingName, value: param.signingName)
+                                sigV4Option.signingProperties.set(key: AttributeKeys.signingRegion, value: param.signingRegionSet?[0])
+                                validAuthOptions.append(sigV4Option)
+                            default:
+                                throw ClientError.authError("Unknown auth scheme name: \(scheme.name)")
+                        }
+                    }
+                    return validAuthOptions
+                }
+
+                public func constructParameters(context: HttpContext) throws -> ClientRuntime.AuthSchemeResolverParameters {
+                    guard let opName = context.getOperation() else {
+                        throw ClientError.dataNotFound("Operation name not configured in middleware context for auth scheme resolver params construction.")
+                    }
+                    guard let endpointParam = context.attributes.get(key: AttributeKey<EndpointParams>(name: "EndpointParams")) else {
+                        throw ClientError.dataNotFound("Endpoint param not configured in middleware context for rules-based auth scheme resolver params construction.")
+                    }
+                    return S3AuthSchemeResolverParameters(operation: opName)
+                }
+            }
+            """.trimIndent()
```

Comment:
nit: Remove `.trimIndent()` and paste the expected text here without adjusting its indentation.  Makes tests easier to update / maintain.

Reply:
Changed as suggested.